### PR TITLE
[Merged by Bors] - style: fix some mis-placed by

### PR DESCRIPTION
**Pull request by grunweg**
---

Commits can be reviewed individually: the second commit is fully automatic (using VS Code's search-and-replace, in regex mode); the last commit fixes the line length (updating the syntax of that comment to Lean 4).

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Calculus/Deriv/Basic.lean [hunk 1]
```diff
@@ -69,8 +69,9 @@ and they more frequently lead to the desired result.
 
 We set up the simplifier so that it can compute the derivative of simple functions. For instance,
 ```lean
-example (x : ℝ) : deriv (fun x ↦ cos (sin x) * exp x) x = (cos(sin(x))-sin(sin(x))*cos(x))*exp(x) :=
-by { simp, ring }
+example (x : ℝ) :
+    deriv (fun x ↦ cos (sin x) * exp x) x = (cos(sin(x))-sin(sin(x))*cos(x))*exp(x) := by
+  simp, ring
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Calculus/Deriv/Basic.lean:74]: But I'm confident that the comma no longer works
```suggestion
  simp; ring
```

File: Mathlib/Analysis/Calculus/Deriv/Basic.lean [hunk 2]
```diff
@@ -69,8 +69,9 @@ and they more frequently lead to the desired result.
 
 We set up the simplifier so that it can compute the derivative of simple functions. For instance,
 ```lean
-example (x : ℝ) : deriv (fun x ↦ cos (sin x) * exp x) x = (cos(sin(x))-sin(sin(x))*cos(x))*exp(x) :=
-by { simp, ring }
+example (x : ℝ) :
+    deriv (fun x ↦ cos (sin x) * exp x) x = (cos(sin(x))-sin(sin(x))*cos(x))*exp(x) := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Calculus/Deriv/Basic.lean:73]: I'm not even sure the over-parenthesized syntax works in lean 4

File: Mathlib/Data/Set/Card.lean [hunk 3]
```diff
@@ -167,8 +167,8 @@ theorem encard_diff_add_encard_inter (s t : Set α) :
     diff_union_inter]
 
 theorem encard_union_add_encard_inter (s t : Set α) :
-    (s ∪ t).encard + (s ∩ t).encard = s.encard + t.encard :=
-by rw [← diff_union_self, encard_union_eq disjoint_sdiff_left, add_right_comm,
+    (s ∪ t).encard + (s ∩ t).encard = s.encard + t.encard := by
+  rw [← diff_union_self, encard_union_eq disjoint_sdiff_left, add_right_comm,
   encard_diff_add_encard_inter]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Set/Card.lean:172]:
```suggestion
    encard_diff_add_encard_inter]
```

File: Mathlib/Tactic/Polyrith.lean [hunk 4]
```diff
@@ -400,18 +400,18 @@ Examples:
 
 ```lean
 example (x y : ℚ) (h1 : x*y + 2*x = 1) (h2 : x = y) :
-  x*y = -2*y + 1 :=
-by polyrith
+  x*y = -2*y + 1 := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Tactic/Polyrith.lean:403]:
```suggestion
    x*y = -2*y + 1 := by
```

File: Mathlib/Tactic/SimpRw.lean [hunk 5]
```diff
@@ -49,8 +49,8 @@ For example, neither `simp` nor `rw` can solve the following, but `simp_rw` can:
 example {a : ℕ}
   (h1 : ∀ a b : ℕ, a - 1 ≤ b ↔ a ≤ b + 1)
   (h2 : ∀ a b : ℕ, a ≤ b ↔ ∀ c, c < a → c < b) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/Tactic/SimpRw.lean:51]:
```suggestion
    (h1 : ∀ a b : ℕ, a - 1 ≤ b ↔ a ≤ b + 1)
    (h2 : ∀ a b : ℕ, a ≤ b ↔ ∀ c, c < a → c < b) :
```

File: Mathlib/Tactic/SimpRw.lean [hunk 6]
```diff
@@ -49,8 +49,8 @@ For example, neither `simp` nor `rw` can solve the following, but `simp_rw` can:
...
-  (∀ b, a - 1 ≤ b) = ∀ b c : ℕ, c < a → c < b + 1 :=
-by simp_rw [h1, h2]
+  (∀ b, a - 1 ≤ b) = ∀ b c : ℕ, c < a → c < b + 1 := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Tactic/SimpRw.lean:52]:
```suggestion
    (∀ b, a - 1 ≤ b) = ∀ b c : ℕ, c < a → c < b + 1 := by
```

File: test/SimpRw.lean [hunk 7]
```diff
@@ -18,21 +18,21 @@ example (f : Nat → Nat) {a b c : Nat} (ha : f b = a) (hc : f b = c) : a = c :=
 example (a b c d : Nat) : a + (b + (c + d)) = ((d + c) + b) + a := by simp_rw [Nat.add_comm]
 
 -- `simp_rw` can also rewrite in assumptions:
-example (p : Nat → Prop) (a b : Nat) (h : p (a + b)) : p (b + a) :=
-by {simp_rw [Nat.add_comm a b] at h; exact h}
+example (p : Nat → Prop) (a b : Nat) (h : p (a + b)) : p (b + a) := by
+  {simp_rw [Nat.add_comm a b] at h; exact h}
```
Comment (Ruben-VandeVelde) [inline test/SimpRw.lean:22]:
```suggestion
  simp_rw [Nat.add_comm a b] at h; exact h
```
and below

File: test/Simps.lean [hunk 8]
```diff
@@ -728,12 +728,12 @@ run_cmd liftTermElabM <| do
...
 
-example (e₁ : α ≃ β) (e₂ : β ≃ γ) (x : α) {z} (h : e₂ (e₁ x) = z) : (e₁.trans e₂) x = z :=
-by simp only [Equiv.trans_apply]; rw [h]
+example (e₁ : α ≃ β) (e₂ : β ≃ γ) (x : α) {z} (h : e₂ (e₁ x) = z) : (e₁.trans e₂) x = z := by
+  simp only [Equiv.trans_apply]; rw [h]
 
 example (e₁ : α ≃ β) (e₂ : β ≃ γ) (x : γ) {z} (h : e₁.symm (e₂.symm x) = z) :
-  (e₁.trans e₂).symm x = z :=
-by simp only [Equiv.trans_symm_apply]; rw [h]
+  (e₁.trans e₂).symm x = z := by
```
Comment (Ruben-VandeVelde) [inline test/Simps.lean:735]:
```suggestion
    (e₁.trans e₂).symm x = z := by
```

**Review by Ruben-VandeVelde — COMMENTED** [review]
You're allowed to argue I'm scope-creeping your PR :)

**Comment by grunweg** [comment]
> You're allowed to argue I'm scope-creeping your PR :)

You are, but that need not be a problem :-) If you test and suggest a change, I'm happy to accept it - I am less willing to do this legwork myself. To this end, I have just accepted most of your suggestions.

File: Mathlib/Analysis/Calculus/Deriv/Basic.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Analysis/Calculus/Deriv/Basic.lean:74]: Oh, indeed!

File: test/SimpRw.lean [hunk 9]
```diff
@@ -18,21 +18,21 @@ example (f : Nat → Nat) {a b c : Nat} (ha : f b = a) (hc : f b = c) : a = c :=
...
 -- or at multiple assumptions:
-example (p : Nat → Prop) (a b : Nat) (h₁ : p (b + a) → p (a + b)) (h₂ : p (a + b)) : p (b + a) :=
-by {simp_rw [Nat.add_comm a b] at h₁ h₂; exact h₁ h₂}
+example (p : Nat → Prop) (a b : Nat) (h₁ : p (b + a) → p (a + b)) (h₂ : p (a + b)) : p (b + a) := by
+  {simp_rw [Nat.add_comm a b] at h₁ h₂; exact h₁ h₂}
```
Comment (grunweg) [inline test/SimpRw.lean:25]:
```suggestion
  simp_rw [Nat.add_comm a b] at h₁ h₂; exact h₁ h₂

File: test/SimpRw.lean [hunk 10]
```diff
@@ -18,21 +18,21 @@ example (f : Nat → Nat) {a b c : Nat} (ha : f b = a) (hc : f b = c) : a = c :=
...
 -- or everywhere:
-example (p : Nat → Prop) (a b : Nat) (h₁ : p (b + a) → p (a + b)) (h₂ : p (a + b)) : p (a + b) :=
-by {simp_rw [Nat.add_comm a b] at *; exact h₁ h₂}
+example (p : Nat → Prop) (a b : Nat) (h₁ : p (b + a) → p (a + b)) (h₂ : p (a + b)) : p (a + b) := by
+  {simp_rw [Nat.add_comm a b] at *; exact h₁ h₂}
```
Comment (grunweg) [inline test/SimpRw.lean:28]:
```suggestion
  simp_rw [Nat.add_comm a b] at *; exact h₁ h₂

**Review by grunweg — COMMENTED** [review]
Good catch on all of them, thanks!

File: Mathlib/Analysis/Calculus/Deriv/Basic.lean (hunk 2 above)
Comment (grunweg) [reply Mathlib/Analysis/Calculus/Deriv/Basic.lean:73]: Trying on the playground: [This syntax parses](https://live.lean-lang.org/#code=import%20Mathlib%0A%0Aopen%20Real%0A%0Aexample%20(x%20%3A%20%E2%84%9D)%20%3A%20deriv%20(fun%20x%20%E2%86%A6%20cos%20(sin%20x)%20*%20exp%20x)%20x%20%3D%20(cos%20(sin%20x)%20-%20sin%20(sin%20x)%20*%20cos%20x)%20*%20exp%20x%20%3A%3D%20by%0A%20%20simp%3B%20ring%0A%0Aexample%20(x%20%3A%20%E2%84%9D)%20%3A%20deriv%20(fun%20x%20%E2%86%A6%20cos%20(sin%20x)%20*%20exp%20x)%20x%20%3D%0A%20%20%20%20(cos%20(sin%20x)%20-%20sin%20(sin%20x)%20*%20cos%20x)%20*%20exp%20x%20%3A%3D%20by%0A%20%20simp%3B%20ring), without the spaces it does not. It seems the example fails, though...

File: test/dfinsupp_notation.lean [hunk 11]
```diff
@@ -1,14 +1,14 @@
 import Mathlib.Data.DFinsupp.Notation
 
-example : (fun₀ | 1 => 3 : Π₀ i, Fin (i + 10)) 1 = 3 :=
-by simp
+example : (fun₀ | 1 => 3 : Π₀ i, Fin (i + 10)) 1 = 3 := by
```
Comment (riccardobrasca) [inline test/dfinsupp_notation.lean:3]: Those can be in the same line as the example, right?

**Review by riccardobrasca — COMMENTED** [review]
Thanks!

bors d+

**Comment by grunweg** [comment]
Thanks for the review; I followed your suggestion.
bors r+

**Pull request closed without merging**